usual/slab_ts.c

```diff
@@ -0,0 +1,76 @@
+#include <usual/slab_ts.h>
+#include <usual/slab.h>
+#include <usual/spinlock.h>
+
+struct ThreadSafeSlab {
+    struct Slab *slab;
+    SpinLock lock;
+};
+
+struct ThreadSafeSlab *thread_safe_slab_create(const char *name, unsigned obj_size, unsigned align,
+                                               slab_init_fn init_func, CxMem *cx) {
+    struct ThreadSafeSlab *ts_slab;
+
+    ts_slab = cx ? cx_alloc0(cx, sizeof(*ts_slab)) : calloc(1, sizeof(*ts_slab));
+    if (!ts_slab)
+        return NULL;
+
+    ts_slab->slab = slab_create(name, obj_size, align, init_func, cx);
+    if (!ts_slab->slab) {
+        free(ts_slab);
+        return NULL;
+    }
+
+    spin_lock_init(&ts_slab->lock);
+    return ts_slab;
+}
+
+void thread_safe_slab_destroy(struct ThreadSafeSlab *ts_slab) {
+    if (!ts_slab)
+        return;
+    spin_lock_acquire(&ts_slab->lock);
+    slab_destroy(ts_slab->slab);
+    spin_lock_release(&ts_slab->lock);
+}
+
+void *thread_safe_slab_alloc(struct ThreadSafeSlab *ts_slab) {
+    void *obj;
+    spin_lock_acquire(&ts_slab->lock);
+    obj = slab_alloc(ts_slab->slab);
+    spin_lock_release(&ts_slab->lock);
+    return obj;
+}
+
+void thread_safe_slab_free(struct ThreadSafeSlab *ts_slab, void *obj) {
+    spin_lock_acquire(&ts_slab->lock);
+    slab_free(ts_slab->slab, obj);
+    spin_lock_release(&ts_slab->lock);
+}
+
+int thread_safe_slab_total_count(struct ThreadSafeSlab *ts_slab) {
```

Comment:
These functions should have function comments, you can probably copy the ones from the slab implementation.